### PR TITLE
cdylib cleanup

### PR DESCRIPTION
**Pull request by folkertdev**

### CODE REVIEW

File: libz-rs-sys-cdylib/zpipe.c [hunk 1]
```diff
@@ -177,13 +198,15 @@ int main(int argc, char **argv)
 {
     int ret;
 
+    fputs("what the hell\n", stderr);
+
```
Comment (bjorn3) [inline libz-rs-sys-cdylib/zpipe.c:202]:
```suggestion
```

File: libz-rs-sys-cdylib/zpipe.c [hunk 2]
```diff
@@ -177,13 +198,15 @@ int main(int argc, char **argv)
...
     /* avoid end-of-line conversions */
     SET_BINARY_MODE(stdin);
     SET_BINARY_MODE(stdout);
 
     /* do compression if no arguments */
     if (argc == 1) {
-        ret = def(stdin, stdout, Z_DEFAULT_COMPRESSION);
+        ret = def(stdin, stdout, Z_BEST_SPEED);
```
Comment (bjorn3) [inline libz-rs-sys-cdylib/zpipe.c:209]: Why was this changed?

File: libz-rs-sys-cdylib/zpipe.c [hunk 3]
```diff
@@ -42,8 +52,14 @@ int def(FILE *source, FILE *dest, int level)
...
-    strm.zalloc = Z_NULL;
-    strm.zfree = Z_NULL;
+#ifdef NO_STD
+    strm.zalloc = custom_zalloc;
+    strm.zfree = custom_zfree;
+#else
+    strm.zalloc = NULL;
+    strm.zfree = NULL;
+#endif
+    fprintf(stderr, "what the hell");
```
Comment (bjorn3) [inline libz-rs-sys-cdylib/zpipe.c:62]:
```suggestion
```

File: libz-rs-sys-cdylib/zpipe.c [hunk 4]
```diff
@@ -42,8 +52,14 @@ int def(FILE *source, FILE *dest, int level)
...
     /* allocate deflate state */
-    strm.zalloc = Z_NULL;
-    strm.zfree = Z_NULL;
+#ifdef NO_STD
+    strm.zalloc = custom_zalloc;
+    strm.zfree = custom_zfree;
+#else
+    strm.zalloc = NULL;
+    strm.zfree = NULL;
+#endif
```
Comment (bjorn3) [inline libz-rs-sys-cdylib/zpipe.c:61]: Unconditionally using custom_zalloc/custom_zfree would be fine too, right?

Comment (folkertdev) [reply libz-rs-sys-cdylib/zpipe.c:61]: it would work, but we use this test for running the various allocators in CI

File: libz-rs-sys-cdylib/zpipe.c (hunk 2 above)
Comment (folkertdev) [reply libz-rs-sys-cdylib/zpipe.c]: no idea how that happened, I'll change that back

**Review by bjorn3 — APPROVED** [review]
